### PR TITLE
Improve git hooks!

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Updated git hooks to only be run when a user pulls AND the `UnrealGDKVersion.bat` (new file) has been updated.
This way we can simply update `UnrealGDKVersion.bat` with new version info when we expect users to need `Setup.bat` and it will be ran automatically.

The post-merge looks like: 
image

Also removed the checkout git hook as not needed.

#### Primary reviewers
@Vatyx 	@danielimprobable

### CODE REVIEW

**Comment by Vatyx** [comment]
Can the version file be changed to match our other versioning files? I don't think it needs to be a bat, no easy way to read what our version is unless you are using batch, so C# scripts can't use it.

**Comment by danielimprobable** [comment]
I'm concerned with the manual updating required of the version file, it would be great if we could find a way of making it automatic to avoid us forgetting to update it.

File: UnrealGDKVersion.bat [hunk 1]
```diff
@@ -0,0 +1,4 @@
+rem Versioning file for the UnrealGDK. 
+rem Updating this file will cause a `git pull` to run Setup.bat for users.
+
+set /p UNREAL_GDK_VERSION=0.1.0
```
[the remark this answers is not in the document]
Comment (joshuahuburn) [reply UnrealGDKVersion.bat]: The reasoning was to allow you to call this bat file and thus get the variable set for you. It's just a minor thing and we can just change it to a text `UnrealGDK.version`

File: UnrealGDK.version [hunk 2]
```diff
@@ -0,0 +1 @@
+0.1.0
```
Comment (girayimprobable) [inline UnrealGDK.version]: Let's call this RequireSetup and just add a counter

File: Setup.bat [hunk 3]
```diff
@@ -10,10 +10,19 @@ call :MarkStartOfBlock "Setup the git hooks"
...
-    echo cmd.exe /c Setup.bat>>.git\hooks\post-checkout
-    echo #!/bin/sh>.git\hooks\post-merge
-    echo cmd.exe /c Setup.bat>>.git\hooks\post-merge
+    rem Remove the old post-checkout hook.
+    if exist .git\hooks\post-checkout del .git\hooks\post-checkout
+
+    rem Remove the old post-merge hook.
+    if exist .git\hooks\post-merge del .git\hooks\post-merge
+
+    rem Add git hook to run Setup.bat when UnrealGDKVersion.bat has been updated.
```
Comment (girayimprobable) [inline Setup.bat]: No `UnrealGDKVersion.bat`

**Pull request merged**